tests/foreman/api/test_settings.py

```diff
@@ -198,6 +198,66 @@ def test_positive_custom_repo_download_policy(setting_update, download_policy, t
     prod.delete()
 
 
+@pytest.mark.run_in_one_thread
+@pytest.mark.parametrize('mirroring_policy', ["mirror_complete"])
+@pytest.mark.parametrize('setting_update', ['default_yum_mirroring_policy'], indirect=True)
+def test_positive_custom_yum_repo_mirroring_policy(setting_update, mirroring_policy, target_sat):
+    """Check the set custom repository yum mirroring policy for newly created yum repository.
+
+    :id: f8901234-5678-4abc-9def-1234567890ab
+
+    :steps:
+        1. Create a product, Organization
+        2. Update the Default YUM Repository mirroring policy in the setting.
+        3. Create a custom yum repo under the created organization.
+        4. Check the set mirroring policy of new created repository.
+
+    :parametrized: yes
+
+    :expectedresults: The set mirroring policy should be the default policy in the newly created
+     yum repository.
+    """
+    org = target_sat.api.Organization().create()
+    prod = target_sat.api.Product(organization=org).create()
+    setting_update.value = mirroring_policy
+    setting_update.update({'value'})
+    repo = target_sat.api.Repository(product=prod, content_type='yum', organization=org).create()
+    assert repo.mirroring_policy == mirroring_policy
+    repo.delete()
+    prod.delete()
```

Comment:
I appreciate doing the cleanup, we don't do that very often. I would suggest to use `request.addfinalizer` so it's executed even if the assertion above fails the test and quits.

Reply:
sounds good :+1: